graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -155,6 +169,8 @@ object Pregel extends Logging {
       i += 1
     }
     messages.unpersist(blocking = false)
+    graphCheckpointer.deleteAllCheckpoints()
+    messageCheckpointer.deleteAllCheckpoints()
```

Comment:
I think when there is an exception during training, if we keep the checkpoints, there is a chance for user to recover from it. I checked in RandomForest/GBT in spark, looks like they only delete the checkpoints when the training successful finished.